README.md
# TYPO3 Extension ``wes_reader``


ChangeLog
(add new changes on top of this file)
1.2.1, 2019-06-12, Weber eBusiness Services <info@weber-ebusiness.de>
    * Add composer.json

1.2.0, 2019-02-25, Weber eBusiness Services <info@weber-ebusiness.de>
    * Add T3 8 and 9 compatibility
    * Remove T3 6.2 compatibility


1.1.1, 2018-07-13, Weber eBusiness Services <info@weber-ebusiness.de>
    * Change ReaderController::getAPISource to use default protocol instead of http


1.1.0, 2016-07-25, Weber eBusiness Services <info@weber-ebusiness.de>
    * Define required settings by using extension configuration
    * Update documentation


1.0.2, 2016-06-15, Weber eBusiness Services <info@weber-ebusiness.de>
    * Update documentation


1.0.1, 2016-06-06, Weber eBusiness Services <info@weber-ebusiness.de>
    * Change naming
    * Remove wes_utils dependency
    * Update documentation


1.0.0, 2016-05-14, Weber eBusiness Services <info@weber-ebusiness.de>
    * Add T3 7 compatibility
    * Remove T3 4.5 compatibility
    * Migrate pibase extension to extbase extension
    * Adapted the new VR 15 API


0.2.7, 2013-11-06, Weber eBusiness Services <info@weber-ebusiness.de>
    * Made loadTCA-call in ext_tables.php conditional


0.2.6, 2013-06-20, Weber eBusiness Services <info@weber-ebusiness.de>
    * Update Comments


0.2.5, 2012-06-21, Weber eBusiness Services <info@weber-ebusiness.de>

    * Updated Popup URL to fit new Linguatec infrastructure
    * Outsourced reader URL into TS setup


0.2.4, 2009-03-13, Weber eBusiness Services <info@weber-ebusiness.de>

    * Added TS option class2hush


0.2.3, 2009-03-13, Weber eBusiness Services <info@weber-ebusiness.de>

    * Modified manual. Thanks to the city of Lebach for sponsoring the creation.
    * Fixed dependency to wes_utils


0.2.2, 2009-03-11, Weber eBusiness Services <info@weber-ebusiness.de>

    * Class name which identifies readable content can be configured in TS setup


0.2.1, 2009-01-16, Weber eBusiness Services <info@weber-ebusiness.de>

    * Extension now uses urlencode() to encode the URL of the current page


0.2.0, 2009-01-12, Weber eBusiness Services <info@weber-ebusiness.de>

    * Added multi language support


0.1.0, 2009-01-08, Weber eBusiness Services <info@weber-ebusiness.de>

    * Initial release


Classes/Controller/ReaderController.php
<?php
namespace WES\WesReader\Controller;

 /***************************************************************
 *  Copyright notice
 *
 *  (c) 2016 Weber eBusiness Services GmbH <info@weber-ebusiness.de>, Weber eBusiness Services GmbH
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/

use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;

class ReaderController extends ActionController
{
    protected static $extConf;

    /**
     * index action
     *
     * @return void
     */
    public function indexAction()
    {
        if (self::getExtConfig('customerid') == '' || self::getExtConfig('typo3codeid') == '') {
            return 'ERROR: No Linguatec customerid or typo3codeid defined. Please define a valid customerid and typo3codeid in the extension configuration.';
        } else {
            $this->view->assignMultiple(
                [
                    'customerid' => trim(self::getExtConfig('customerid')),
                    'api' => $this->getAPISource(self::getExtConfig('typo3codeid')),
                ]
            );
        }
    }

    private static function getExtConfig($key)
    {
        if (!self::$extConf) {
            self::$extConf = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['wes_reader']);
        }
        return self::$extConf[$key];
    }

    private function getAPISource($id)
    {
        return '//vrweb15.linguatec.org/VoiceReaderWeb15User/player/scripts/'.trim($id).'.js';
    }
}


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "wes_reader".
 *
 * Auto generated 06-11-2013 15:41
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = [
    'title' => 'Voice Reader Text-to-Speech',
    'description' => 'Official extension for integrating an online screen reader using Linguatec Voice Reader Web (www.linguatec.de). Can be used on any site with html content. Requires a valid Voice Reader Web license. See manual for a free trial license. Keywords: Text to Speech, Sprachausgabe, Sprachsynthese, Vorlesesoftware, Reader, Vorlesen, Speech, Accessibility, Inclusion, Inklusion, Barrierefreiheit.',
    'category' => 'plugin',
    'author' => 'Weber eBusiness Services',
    'author_email' => 'info@weber-ebusiness.de',
    'author_company' => 'Weber eBusiness Services GmbH',
    'shy' => '',
    'dependencies' => '',
    'conflicts' => '',
    'priority' => '',
    'module' => '',
    'state' => 'stable',
    'internal' => '',
    'uploadfolder' => 0,
    'createDirs' => '',
    'modify_tables' => '',
    'clearCacheOnLoad' => 0,
    'lockType' => '',
    'version' => '1.2.1',
    'constraints' =>
    [
        'depends' =>
        [
            'typo3' => '7.6.0-9.99.99',
            'extbase' => '7.6.0-9.99.99',
            'fluid' => '7.6.0-9.99.99',
        ],
        'conflicts' => [],
        'suggests' => [],
    ],
    'suggests' => [],
];
